src/Sulu/Bundle/ContentBundle/Resources/public/js/components/smart-content/categories/main.js

```diff
@@ -0,0 +1,256 @@
+            this.$el.html(
+                this.templates.skeleton(
+                    {
+                        constants: constants,
+                        operatorLabel: this.translations.operatorLabel,
+                        categoriesLabel: this.translations.categoriesLabel
+                    }
+                )
+            );
+        },
+
+        /**
+         * Init sub-components operator select and datagrid for categories.
+         */
+        initComponents: function() {
+            this.sandbox.start(
+                [
+                    {
+                        name: 'select@husky',
+                        options: {
+                            el: this.$find('.' + constants.operatorClass),
+                            instanceName: this.options.instanceName,
+                            value: 'name',
+                            data: [
+                                {id: operators.or, name: this.translations.useAnyCategory},
+                                {id: operators.and, name: this.translations.useAllCategories}
+                            ],
+                            preSelectedElements: [operators[this.data.operator]]
+                        }
+                    },
+                    {
+                        name: 'datagrid@husky',
+                        options: {
+                            el: this.$find('.' + constants.categoriesClass),
+                            instanceName: this.options.instanceName,
+                            url: '/admin/api/categories?flat=true&sortBy=depth&sortOrder=asc',
+                            resultKey: 'categories',
+                            pagination: false,
+                            childrenPropertyName: 'hasChildren',
+                            resizeListener: false,
+                            selectedCounter: false,
+                            preselected: this.data.ids,
+                            viewOptions: {
+                                table: {
+                                    noItemsText: this.translations.noCategoriesAvailable,
+                                    showHead: false,
+                                    cssClass: 'white-box',
+                                    selectItem: {type: 'checkbox', inFirstCell: true}
+                                }
+                            },
+                            matchings: [
+                                {name: 'name', content: 'Name'},
+                                {name: 'id', disabled: true},
+                                {name: 'children', disabled: true},
+                                {name: 'parent', disabled: true}
+                            ]
+                        }
+                    }
+                ]
+            ).then(
+                function() {
+                    this.sandbox.once(
+                        this.sandbox.events.createEventName('husky.datagrid.', 'view.rendered', this.options.instanceName),
```

Comment:
maybe we could introduce a global namespace and eventname registry? /cc @danrot @chirimoya